### PR TITLE
Add missing cli options for kube-controller-manager and kube-scheduler

### PR DESCRIPTION
**Pull request by Evalle**
PR Summary:
- Adds missing cli options for kube-controller-manager and kube-scheduler
- Fixes https://github.com/kubernetes/kops/issues/9695

### CODE REVIEW

**Comment by hakman** [comment]
@Evalle Thanks for taking this. Please run `make apimachinery` and probably also `make crds` and commit the result. These should fix the tests.

**Comment by Evalle** [comment]
Hi @hakman thanks for the review! 
I actually run `make apimachinery crds` prior to commit

**Comment by hakman** [comment]
Ah... right. I think there is something broken there. Please run `hack/make-apimachinery.sh` instead.

**Comment by Evalle** [comment]
It didn't make any changes 
```
❯ pwd
/Projects/kops
❯ ./hack/make-apimachinery.sh
go: downloading k8s.io/code-generator v0.18.6
go: downloading github.com/spf13/pflag v1.0.5
go: downloading k8s.io/gengo v0.0.0-20200710205751-c0d492a0f3ca
go: downloading k8s.io/klog v1.0.0
go: downloading k8s.io/klog/v2 v2.0.0
go: downloading golang.org/x/tools v0.0.0-20200626171337-aa94e735be7f
go: downloading github.com/go-logr/logr v0.1.0
go: downloading golang.org/x/mod v0.3.0
go: downloading golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543
golang.org/x/mod/semver
golang.org/x/xerrors/internal
github.com/go-logr/logr
golang.org/x/xerrors
golang.org/x/tools/go/ast/astutil
golang.org/x/tools/internal/event/label
golang.org/x/tools/internal/event/keys
github.com/spf13/pflag
golang.org/x/mod/module
golang.org/x/tools/internal/event/core
golang.org/x/tools/internal/fastwalk
golang.org/x/tools/internal/event
golang.org/x/tools/internal/gopathwalk
golang.org/x/tools/internal/gocommand
k8s.io/gengo/types
k8s.io/gengo/namer
k8s.io/klog/v2
k8s.io/klog
k8s.io/gengo/parser
k8s.io/code-generator/pkg/util
golang.org/x/tools/internal/imports
golang.org/x/tools/imports
k8s.io/gengo/generator
k8s.io/gengo/args
k8s.io/code-generator/cmd/conversion-gen/args
k8s.io/code-generator/cmd/conversion-gen/generators
k8s.io/code-generator/cmd/conversion-gen
❯ echo $?
0
❯ git status
On branch ISSUE-9695
Your branch is up to date with 'origin/ISSUE-9695'.

nothing to commit, working tree clean
```

**Comment by hakman** [comment]
May be related to Go version (using 1.15rc2 at the moment) or GOPATH. You should get something like this from `make apimachinery`:
```diff
--- a/pkg/apis/kops/v1alpha2/zz_generated.conversion.go
+++ b/pkg/apis/kops/v1alpha2/zz_generated.conversion.go
@@ -4003,6 +4003,8 @@ func autoConvert_v1alpha2_KubeControllerManagerConfig_To_kops_KubeControllerMana
        out.ConcurrentResourceQuotaSyncs = in.ConcurrentResourceQuotaSyncs
        out.ConcurrentServiceaccountTokenSyncs = in.ConcurrentServiceaccountTokenSyncs
        out.ConcurrentRcSyncs = in.ConcurrentRcSyncs
+       out.AuthenticationKubeconfig = in.AuthenticationKubeconfig
+       out.AuthorizationKubeconfig = in.AuthorizationKubeconfig
        out.EnableProfiling = in.EnableProfiling
        return nil
 }
@@ -4063,6 +4065,8 @@ func autoConvert_kops_KubeControllerManagerConfig_To_v1alpha2_KubeControllerMana
        out.ConcurrentResourceQuotaSyncs = in.ConcurrentResourceQuotaSyncs
        out.ConcurrentServiceaccountTokenSyncs = in.ConcurrentServiceaccountTokenSyncs
        out.ConcurrentRcSyncs = in.ConcurrentRcSyncs
+       out.AuthenticationKubeconfig = in.AuthenticationKubeconfig
+       out.AuthorizationKubeconfig = in.AuthorizationKubeconfig
        out.EnableProfiling = in.EnableProfiling
        return nil
 }
@@ -4212,6 +4216,8 @@ func autoConvert_v1alpha2_KubeSchedulerConfig_To_kops_KubeSchedulerConfig(in *Ku
        out.MaxPersistentVolumes = in.MaxPersistentVolumes
        out.Qps = in.Qps
        out.Burst = in.Burst
+       out.AuthenticationKubeconfig = in.AuthenticationKubeconfig
+       out.AuthorizationKubeconfig = in.AuthorizationKubeconfig
        out.EnableProfiling = in.EnableProfiling
        return nil
 }
@@ -4239,6 +4245,8 @@ func autoConvert_kops_KubeSchedulerConfig_To_v1alpha2_KubeSchedulerConfig(in *ko
        out.MaxPersistentVolumes = in.MaxPersistentVolumes
        out.Qps = in.Qps
        out.Burst = in.Burst
+       out.AuthenticationKubeconfig = in.AuthenticationKubeconfig
+       out.AuthorizationKubeconfig = in.AuthorizationKubeconfig
        out.EnableProfiling = in.EnableProfiling
        return nil
 }
```

**Comment by Evalle** [comment]
interesting, I have `1.14.5`
```
❯ go version
go version go1.14.5 darwin/amd64
```
Let me try with `1.14.6` or `1.15.x`

**Comment by Evalle** [comment]
I think I've found the issue :) it was related to GOPATH indeed, I had kops in both `/go/src` and in `Projects` directory.

**Comment by hakman** [comment]
@Evalle would `authorization-always-allow-paths` also be needed, or is `/healthz`enough?
```
--authorization-always-allow-paths stringSlice     Default: [/healthz]
  A list of HTTP paths to skip during authorization, i.e. these are authorized without contacting the 'core' kubernetes server.
```

**Comment by Evalle** [comment]
Hm, I would ask @rifelpet about this

**Comment by hakman** [comment]
And also @a8j8i8t8: https://github.com/kubernetes/kops/pull/9726#issuecomment-671863343.

File: pkg/apis/kops/componentconfig.go [hunk 1]
```diff
@@ -581,6 +581,10 @@ type KubeControllerManagerConfig struct {
 	ConcurrentServiceaccountTokenSyncs *int32 `json:"concurrentServiceaccountTokenSyncs,omitempty" flag:"concurrent-serviceaccount-token-syncs"`
 	// The number of replicationcontroller objects that are allowed to sync concurrently.
 	ConcurrentRcSyncs *int32 `json:"concurrentRcSyncs,omitempty" flag:"concurrent-rc-syncs"`
+	// AuthenticationKubeconfig is the path to a Authentication Kubeconfig
```
Comment (rifelpet) [inline pkg/apis/kops/componentconfig.go:584]: nit:

```suggestion
	// AuthenticationKubeconfig is the path to an Authentication Kubeconfig
```

and similar for below...

Comment (Evalle) [reply pkg/apis/kops/componentconfig.go]: fixed

**Comment by hakman** [comment]
@Evalle Please add also the `--authorization-always-allow-paths` if that's ok with you.

**Comment by Evalle** [comment]
Yes, I'm working on it ATM :)

**Comment by hakman** [comment]
Thanks :)

**Comment by Evalle** [comment]
Should I squash commits ?

**Comment by hakman** [comment]
Looks pretty good to me. Congrats on your first Kops PR :)
/lgtm
/approve

**Comment by Evalle** [comment]
Happy to help :) thanks for the review @hakman

**Comment by hakman** [comment]
My pleasure! 👍

**Pull request merged**